src/ui/public/share/directives/share_object_url.js

```diff
@@ -69,8 +70,14 @@ app.directive('shareObjectUrl', function (Private, Notifier) {
         });
       };
 
+      // since getUrl() is called within a watcher we cache the unhashing step
+      const unhashStatesInAbsUrl = memoize((absUrl) => {
+        return Private(UnhashStatesProvider).inAbsUrl(absUrl);
+      });
+
       $scope.getUrl = function () {
-        let url = $location.absUrl();
+        let url = unhashStatesInAbsUrl($location.absUrl());
```

Comment:
I think renaming this to `const unhashedUrl` will have the same readability benefit I mentioned above. I think using it the same way, with the same variable names, in similar contexts, also helps create a usage pattern which is easier to recognize throughout the code.